pkg/cmd/ssh_gcp.go

```diff
@@ -71,9 +77,15 @@ func sshToGCPNode(nodeName, path, user, pathSSKeypair string, sshPublicKey []byt
 	time.Sleep(45 * time.Second)
 
 	key := filepath.Join(pathSSKeypair, "key")
-	sshCmd := fmt.Sprintf("ssh -i " + key + " -o \"ProxyCommand ssh -W %%h:%%p -i " + key + " -o IdentitiesOnly=yes -o StrictHostKeyChecking=no " + bastionNode + "\" " + node + " -o IdentitiesOnly=yes -o StrictHostKeyChecking=no")
+	var sshCmd string
+	if debugSwitch {
+		sshCmd = fmt.Sprintf("ssh -v -i " + key + " -o \"ProxyCommand ssh -W %%h:%%p -i " + key + " -o IdentitiesOnly=yes -o StrictHostKeyChecking=no " + bastionNode + "\" " + node + " -o IdentitiesOnly=yes -o StrictHostKeyChecking=no")
+	} else {
+		sshCmd = fmt.Sprintf("ssh -i " + key + " -o \"ProxyCommand ssh -W %%h:%%p -i " + key + " -o IdentitiesOnly=yes -o StrictHostKeyChecking=no " + bastionNode + "\" " + node + " -o IdentitiesOnly=yes -o StrictHostKeyChecking=no")
+	}
+
 	fmt.Println(sshCmd)
-	cmd := exec.Command("bash", "-c", sshCmd)
+	cmd := exec.Command(sshCmd)
```

Comment:
Have you tried it? This does not work. The signature of exec.Command is `Command(name string, arg ...string)`, hence the first argument would be ssh and the rest is put in as args

When you have the args as array, you can append the debug flag without the need to duplicate the command. This is how I would do it:
```go
	args := []string{"-i" + key, "-oProxyCommand=ssh -W%h:%p -i" + key + " -oIdentitiesOnly=yes -oStrictHostKeyChecking=no " + bastionNode, node, "-oIdentitiesOnly=yes", "-oStrictHostKeyChecking=no"}
	if debugSwitch {
		args = append([]string{"-vvv"}, args...)
	}
	fmt.Println("ssh " + strings.Join(args[:], " "))
	cmd := exec.Command("ssh", args...)
```
You could also pass a debug flag to the ssh command in the proxy command argument, I guess

Reply:
Thanks, I like this tips. updated, and did some testing
```args = append([]string{"-vvv"}, append(args[:2], append([]string{"-vvv"}, args[2:]...)...)...)``` , but unfortunately not working somehow.  
```
[-vvv -i/Users/i333878/.garden/cache/canary-virtual/projects/i333878/b68cklqfl1/key -oProxyCommand=ssh -W%h:%p -vvv  -i/Users/i333878/.garden/cache/canary-virtual/projects/i333878/b68cklqfl1/key -oIdentitiesOnly=yes -oStrictHostKeyChecking=no gardener@34.77.207.104 gardener@shoot--i333878--b68cklqfl1-worker-b5hoj-z1-c467bc486-jqsg6 -oIdentitiesOnly=yes -oStrictHostKeyChecking=no]

debug1: identity file /Users/i333878/.garden/cache/canary-virtual/projects/i333878/b68cklqfl1/key-cert type -1
debug1: Local version string SSH-2.0-OpenSSH_8.1
usage: ssh [-46AaCfGgKkMNnqsTtVvXxYy] [-B bind_interface]
           [-b bind_address] [-c cipher_spec] [-D [bind_address:]port]
           [-E log_file] [-e escape_char] [-F configfile] [-I pkcs11]
           [-i identity_file] [-J [user@]host[:port]] [-L address]
           [-l login_name] [-m mac_spec] [-O ctl_cmd] [-o option] [-p port]
           [-Q query_option] [-R address] [-S ctl_path] [-W host:port]
           [-w local_tun[:remote_tun]] destination [command]
kex_exchange_identification: Connection closed by remote host
```

Reply:
@tedteng if you want to add the verbose flag also to the ProxyCommand you have to do something like this

```go
	proxyCommandArgs := []string{"-W%h:%p", "-i" + key, "-oIdentitiesOnly=yes", "-oStrictHostKeyChecking=no", bastionNode}
	if debugSwitch {
		proxyCommandArgs = append([]string{"-vvv"}, proxyCommandArgs...)
	}
	args := []string{"-i" + key, "-oProxyCommand=ssh " + strings.Join(proxyCommandArgs[:], " "), node, "-oIdentitiesOnly=yes", "-oStrictHostKeyChecking=no"}
	if debugSwitch {
		args = append([]string{"-vvv"}, args...)
	}

	fmt.Println("ssh " + strings.Join(args[:], " "))
	cmd := exec.Command("ssh", args...)
```

Reply:
give it up, . It seems can't fit in one line.  Thanks, It works now